# On Android, use system valgrind, instead of ones checked in to the Chromium
# repository.

--- a/base/third_party/valgrind/memcheck.h
+++ b/base/third_party/valgrind/memcheck.h
@@ -1,4 +1,6 @@
-
+#ifdef ANDROID
+  #include "memcheck/memcheck.h"
+#else
 /*
    ----------------------------------------------------------------
 
@@ -277,3 +279,4 @@ typedef
 
 #endif
 
+#endif
--- a/base/third_party/valgrind/valgrind.h
+++ b/base/third_party/valgrind/valgrind.h
@@ -1,3 +1,6 @@
+#ifdef ANDROID
+  #include "include/valgrind.h"
+#else
 /* -*- c -*-
    ----------------------------------------------------------------
 
@@ -4790,3 +4793,5 @@ VALGRIND_PRINTF_BACKTRACE(const char *fo
 #undef PLAT_ppc64_aix5
 
 #endif   /* __VALGRIND_H */
+
+#endif
